tests/features/save_test.py

```diff
@@ -292,15 +294,19 @@ def test_save_and_exit_to_uncreated_directory(run, tmpdir):
     assert f.read() == 'hello\n'
 
 
-def test_save_and_exit_to_dot_file(run, tmpdir):
-    f = tmpdir.join('.f')
-    with run(str(f)) as h:
-        h.press('hello')
-        h.await_text('hello')
-        h.press_and_enter('^X')
-        h.await_text('file is modified - save [yes, no]?')
-        h.press('y')
-        h.await_text('enter filename: ')
-        h.press('Enter')
-        h.await_exit()
-    assert f.read() == 'hello\n'
+@pytest.mark.parametrize('dot_file', ('.f',))
+def test_save_and_exit_to_dot_file(run, dot_file):
+    with open(dot_file, 'w+') as f:
+        with run(dot_file) as h:
+            h.press('hello')
+            h.await_text('hello')
+            h.press_and_enter('^X')
+            h.await_text('file is modified - save [yes, no]?')
+            h.press('y')
+            h.await_text('enter filename: ')
+            h.press('Enter')
+            h.await_exit()
+        try:
+            assert f.read() == 'hello\n'
+        finally:
+            os.remove(os.path.abspath(dot_file))
```

Comment:
this leaves the file around if an assertion fails above which is very undesirable

Reply:
Nope, the file gets deleted no matter if the assertion fails or not. As you can see in the example where i run `pytest tests/features/save_test.py -k "test_save_and_exit_to_dot_file" ; ll` the file seems to get deleted correctly, but correct me if i'm wrong.
![dot-file-test](https://user-images.githubusercontent.com/94523763/151248977-19fc3eb0-16d0-408d-88b5-010048e30bcf.png)

Reply:
key word *above* -- you're not protecting the full critical path where this file could be written